src/mpid/ch4/netmod/ofi/ofi_huge.c

```diff
@@ -384,7 +384,8 @@ int MPIDI_OFI_huge_chunk_done_event(int vni, struct fi_cq_tagged_entry *wc, void
     MPIR_cc_decr(chunk_req->chunks_outstanding, &c);
 
     if (c == 0) {
-        MPL_free(chunk_req->chunks_outstanding);
+        /* cast to suppress warning -- discard volatile qualifier */
+        MPL_free((void *) chunk_req->chunks_outstanding);
```

Comment:
I am really tempted to say remove the volatile qualifier here:
```
src/include/mpir_refcount_global.h:typedef volatile int MPIR_cc_t;
```
Since we have locks protecting access to completion counters in global mode. IMO, it is a misuse of `volatile`.

Reply:
From https://en.cppreference.com/w/c/language/volatile:

> Note that volatile variables are not suitable for communication between threads; they do not offer atomicity, synchronization, or memory ordering. A read from a volatile variable that is modified by another thread without synchronization or concurrent modification from two unsynchronized threads is undefined behavior due to a data race.

Reply:
With global thread-cs, we always read and modify the cc counter inside the global critical sections, right? So we aren't accessing the variable without synchronization or having concurrent modifications. `volatile` prevents compiler optimizations, which seems plausible.

Reply:
I proposed the removal in #5642. We can discuss it separately, if needed.